src/legacy/core_plugins/embeddable_api/public/actions/action.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiContextMenuItemIcon } from '@elastic/eui';
+import { trackUiMetric } from '../../../ui_metric/public';
+
+import { IEmbeddable } from '../embeddables';
+
+export interface ExecuteActionContext<
+  TEmbeddable extends IEmbeddable = IEmbeddable,
+  TActionContext extends {} = {}
+> {
+  embeddable: TEmbeddable;
+  triggerContext?: TActionContext;
+}
+
+export interface ActionContext<TEmbeddable extends IEmbeddable = IEmbeddable> {
+  embeddable: TEmbeddable;
+}
+
+export abstract class Action<
+  TEmbeddable extends IEmbeddable = IEmbeddable,
+  TActionContext extends {} = {}
+> {
+  /**
+   * Determined the order when there is more than one action matched to a trigger.
+   * Higher numbers are displayed first.
+   */
+  public order: number = 0;
+  public abstract readonly type: string;
+
+  constructor(public readonly id: string) {
+    trackUiMetric('EmbeddableAPI', `actionConstructor`);
```

Comment:
Yea, I think for now I'll leave as is.  A good next step will be getting this embeddable_api plugin prepped for NP migration... I don't think there is too much to it so it could be a good first contender.